src/Components/Components/src/RenderTree/RenderTreeBuilder.cs

```diff
@@ -454,13 +454,30 @@ public void OpenComponent(int sequence, Type componentType)
         /// <param name="value">The value for the key.</param>
         public void SetKey(object value)
         {
-            // This is just a placeholder to enable work in parallel in the
-            // aspnetcore-tooling repo.
-            //
-            // The real implementation will involve multiple overloads, likely:
-            //   SetKey(int value) -- underlying logic
-            //   SetKey<T>(T value) where T: struct -- avoids boxing 'value' before calling .GetHashCode()
-            //   SetKey(object value) -- performs null check before calling .GetHashCode()
```

Comment:
In the end, there's no need for the anticipated overloads, because we're now storing the key as an `object` instead of an `int`-valued hashcode.